### PR TITLE
(PUP-10299) Add SemVerRange support to zypper

### PR DESCRIPTION
**Pull request by ciprianbadescu**
Add support of >, >=, <, <=, >=A <=B ranges for package versions

### CODE REVIEW

File: lib/puppet/provider/package/zypper.rb [hunk 1]
```diff
@@ -50,6 +50,38 @@ def zypper_version
...
+      rescue Puppet::Util::Package::Version::Range::ValidationFailure, Puppet::Util::Package::Version::Rpm::ValidationFailure
+        Puppet.debug("Cannot parse #{should} as a RPM version range")
+        return should
+      end
+
+      sorted_versions = SortedSet.new
+
+      output = zypper 'search', '--match-exact', '--type', 'package', '--uninstalled-only', '-s', @resource[:name]
+      output.lines.each do |line|
+
```
Comment (gimmyxd) [inline lib/puppet/provider/package/zypper.rb:66]: nit: extra blank line

**Comment by ciprianbadescu** [comment]
jenkins please test this on sles11-64a,sles11-32a,sles12-64a,sles12-POWERa,sles15-64a

**Comment by ciprianbadescu** [comment]
jenkins please test this on sles11-64a,sles11-32a,sles12-64a,sles12-POWERa,sles15-64a

File: acceptance/tests/provider/package/zypper_install_packet_with_range.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+test_name "zypper can install range if package is not installed" do
```
Comment (GabrielNagy) [inline acceptance/tests/provider/package/zypper_install_packet_with_range.rb:1]: nit: typo in filename - packet/package

**Pull request merged**